server/src/main/java/org/opensearch/transport/NetworkMessage.java

```diff
@@ -48,7 +48,7 @@ public abstract class NetworkMessage {
     protected final long requestId;
     protected final byte status;
 
-    NetworkMessage(ThreadContext threadContext, Version version, byte status, long requestId) {
+    public NetworkMessage(ThreadContext threadContext, Version version, byte status, long requestId) {
```

Comment:
This constructor could be kept package private (unchanged) and moved to `nativeprotocol` package: it is used only by native protocol